### DISCUSSION TITLE
Pass command line arguments to userscripts

### DISCUSSION THREAD

**Discussion by towalker** [qutebrowser/qutebrowser#5772] [Q&A] [answered]
I've been creating a few userscripts in Python, but I'm not being able to pass arguments from the command line to them. An example would be:
`:spawn --userscript scriptname argument1 argument2`

I thought the system variable QUTE_COMMANDLINE_TEXT would contain everything in this line, however it seems to be always empty. Do you know what am I getting wrong?

`command_line = os.environ.get('QUTE_COMMANDLINE_TEXT')`

Also, my compliments to The-Compiler and the community built around qutebrowser. It is a great tool and I'm happy to finally have access to this level of customization through scripting in a web browser.

**Comment by emanuele6** [comment, marked as answer]
You need to use the commandline arguments of your userscript program.
In `python`, you get these values from the `sys.argv` array in the `sys` module.

**Reply by The-Compiler** [reply]
Yep - or use `argparse` if you have something more sophisticated with flags and such.

@towalker `QUTE_COMMANDLINE_TEXT` is only useful if you want to modify the text in the commandline at the moment the userscript is launched - you could write an userscript invoked via a keybinding in command mode which reads the current text and somehow changes it by using the `:set-cmd-text` command.

**Reply by towalker** [reply]
Thank you emanauele6 and The-Compiler. I was so focused on the qutebrowser environment variables that I forgot to try the usual way of passing arguments.
